build.gradle.kts

```diff
@@ -0,0 +1,210 @@
+    compile("com.adobe.xmp", "xmpcore", "5.1.2")
+    compile("com.drewnoakes", "metadata-extractor", "2.9.1")
+    compile("net.objecthunter", "exp4j", "0.4.8")
+    compile("org.atteo", "evo-inflector", "1.2.2")
+    compile("com.thoughtworks.xstream", "xstream", "1.4.10")
+
+    //	compile("com.twelvemonkeys.imageio", "imageio-core", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-bmp", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-jpeg", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-iff", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-icns", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pcx", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pict", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-clippath", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-hdr", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pdf", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pnm", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-psd", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tga", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-sgi", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-thumbsdb", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tiff", "3.3.2")
+
+    compile(files(file("extra/lib").listFiles()))
+
+}
+
+// source directories
+java.sourceSets {
+    getByName("main") {
+        java.srcDir("src")
+        resources.srcDir("src")
+    }
+}
+
+tasks {
+    val main = "_Main"
+
+    withType<JavaCompile> {
+        options.isWarnings = false
+        options.isDeprecation = false
+        // javac args
+        options.compilerArgs = listOf("-Xlint:unchecked", "-nowarn",
+                "--add-exports", "javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.javafx.scene.traversal=ALL-UNNAMED",
+                "--add-exports", "javafx.web/com.sun.webkit=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.glass.ui=ALL-UNNAMED")
+    }
+
+    "compileJava" {
+        doFirst {
+            println("Java version: ${JavaVersion.current()}")
+        }
+    }
+
+    "compileKotlin" {
+        doFirst {
+            println("Kotlin version: $kotlinVersion")
+        }
+    }
+
+    "build" {
+        group = main
+    }
+
+    "run" {
+        group = main
+    }
+
+    val cleanup by creating {
+        group = main
+        description = "Cleans the working dir - currently only deletes logs"
+        doFirst {
+            println("Cleaning up...")
+            file("working dir/log").listFiles { file -> arrayOf("log", "zip").contains(file.extension) }.forEach { it.delete() }
+        }
+    }
+
+    val copyLibs by creating(Copy::class) {
+        group = main
+        description = "Copies all libraries into the working dir for deployment"
```

Comment:
Whjy are the local libs moved to extra/lib and then copied if they can be in their targed directory from the beginning? THis only makes the build slower and more complex. Makes no sense to me.

Reply:
1. extra/libs is going to be removed in the future, once updates are available
2. only the binaries should be included in the build, not sources and javadoc

Reply:
For the application to run it needs those libs in the working directory though. I'd like for the development/build dir structure to be as close to the deployment dir structure as possible. So that minimal effort on the part of user as well as gradle be necessary - less things to go wrong, less need to understand what the build does, less things to explain about the build.

1 What do you mean by updates?
2 Normally I would agree, but this is portable application here.

Reply:
1. with updates I meant new versions of the libraries that are not yet pulled from remote
2. Why would it need sources and javadoc if its portable? Being a portable application it should all the more be as small as possible, and sources would definitely bloat it

I just realised that I made a little mistake, fixed it with commit 9be21db0df8d138c64d442bacb40e95173469dbd

Reply:
2 All dependent libs must be on classpath for the app to run, in our case in the working_dir/lib directory. The sources/javadoc do not have to be removed (this can be done in the cleanup stage of the deploy task), but I do not want. Id like the app to always run from the working_dir even in development, using the same structure as if it was deployed, with the added benefit of deployment only requiring to copy the whole working directory somewhere else and with a clean up.

Reply:
It does run from working dir even while developing. The libs aren't deleted after running, so after running it once, it will be runnable always. The important thing here is that it also copies the libs from all other dependencies which aren't explicitly included in the project anymore but managed by gradle.